dotnet/src/SemanticKernel.Abstractions/Memory/MemoryRecordAttributes/KeyAttribute.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel.Memory;
+
+/// <summary>
+/// Attribute to mark a property on a class as the key under which data is stored in a vector store.
+/// </summary>
+[Experimental("SKEXP0001")]
+[AttributeUsage(AttributeTargets.Property, AllowMultiple = false)]
+public sealed class KeyAttribute : Attribute
```

Comment:
Any concerns that this generic attribute name could conflict (same for others)?

Should we prefix the attribute names with `MemoryRecord`?

Reply:
These will certainly clash.  There are similar attributes in a number of other libraries.  Both EF and the Odata libraries e.g. have a KeyAttribute.  I had what you are proposing first but the length of the name wasn't great.  That said, it's a tradeoff between having the long names, and requiring people to re-map the class names if there are clashes which also sucks.  Happy to go with the consensus here.

Reply:
@westey-m Specifically for `Key` attribute, can we use existing one from `System.ComponentModel.DataAnnotations`? 
https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.keyattribute

We can still keep `VectorAttribute` and others as new ones on our side, but re-use existing ones which we get out of the box in .NET.

Reply:
@dmytrostruk, I am concerned that we may need to support additional properties on the Key and other attributes to support create and filtered search scenarios.  E.g. Azure AI Search has many options you can set on a Key, see https://learn.microsoft.com/en-us/azure/search/search-get-started-vector#create-a-vector-index
If any of these scenarios make sense across most of the databases, it would make sense for us to support them too.

Having our own attribute will allow us to add these as/if needed without requiring users to switch to a new type.

I've updated the doc with my reasoning as well, but let's chat more directly later.

Reply:
Makes sense, thank you!